### ISSUE TITLE
502 Bad Gateway / nginx Connection refused

### ISSUE THREAD

**Issue by edemaine** [zodern/meteor-up#1124]
Mup version (`mup --version`): 1.5.0-beta.5 on Node v12.13.1

Mup config

```js
{
  "servers": {
    "one": {
      "host": "1.2.3.4",
      "username": "ubuntu",
      "pem": "~/.ssh/pem"
    }
  },
  "mongo": {
    "oplog": true,
    "port": 27017,
    "servers": {
      "one": {}
    },
    "version": "3.4.1"
  },
  "proxy": {
    "domains": "host1,host2",
    "ssl": {
      "forceSSL": true,
      "letsEncryptEmail": "edemaine@mit.edu"
    },
    "clientUploadLimit": "0"
  },
  "hooks": {
    "pre.deploy": {
      "localCommand": "npm install"
    }
  },
  "app": {
    "name": "my-app",
    "path": "/afs/csail/u/e/edemaine/Projects/coinsliding",
    "servers": {
      "one": {}
    },
    "docker": {
      "image": "abernix/meteord:node-8.4.0-base",
      "stopAppDuringPrepareBundle": true,
      "imagePort": 3000,
      "args": [
        "--link=mongodb:mongodb"
      ]
    },
    "buildOptions": {
      "serverOnly": true,
      "buildLocation": "/scratch/build"
    },
    "env": {
      "ROOT_URL": "https://subdomain.host.com",
      "MONGO_URL": "mongodb://mongodb:27017/my-app",
      "MONGO_OPLOG_URL": "mongodb://mongodb/local",
      "VIRTUAL_HOST": "host1,host2",
      "HTTPS_METHOD": "redirect",
      "LETSENCRYPT_HOST": "host1,host2",
      "LETSENCRYPT_EMAIL": "edemaine@mit.edu",
      "VIRTUAL_PORT": 3000,
      "HTTP_FORWARDED_COUNT": 1
    },
    "deployCheckWaitTime": 200,
    "type": "meteor"
  }
}
```

After `mup setup`/`mup deploy`, I get a 502 Bad Gateway (once LetsEncrypt certificates get created).

Output of `mup proxy logs`:
```bash
nginx.1    | 2020/01/15 20:33:09 [error] 192#192: *41 connect() failed (111: Connection refused) while connecting to upstream, client: 149.28.73.26, server: coinsliding.erikdemaine.org, request: "GET / HTTP/1.1", upstream: "http://172.17.0.5:80/", host: "coinsliding.erikdemaine.org"
nginx.1    | coinsliding.erikdemaine.org 149.28.73.26 - - [15/Jan/2020:20:33:09 +0000] "GET / HTTP/1.1" 502 157 "-" "downnotifier.com monitoring"
nginx.1    | 2020/01/15 20:34:14 [error] 193#193: *43 connect() failed (111: Connection refused) while connecting to upstream, client: 104.236.216.73, server: coinsliding.erikdemaine.org, request: "GET / HTTP/1.1", upstream: "http://172.17.0.5:80/", host: "coinsliding.erikdemaine.org"
nginx.1    | coinsliding.erikdemaine.org 104.236.216.73 - - [15/Jan/2020:20:34:14 +0000] "GET / HTTP/1.1" 502 157 "-" "downnotifier.com monitoring"
```

Past reports of this issue suggest that the server never starts.  However, `mup logs` gives messages from my app (Kadira enabled messages) suggesting it started just fine.  As further evidence, I just tried the exact same configuration on mup 1.4.6 on Node v6.11.4 (the only Node where I can authenticate with old mup), and the proxy worked just fine.  So perhaps there is an issue with the latest beta?

**Comment by zodern** [comment]
Could you please run `mup meteor status` and check what the value of the `PORT` env variable is? It should be 3000 for everything to work correctly.

**Comment by edemaine** [comment]
`mup meteor status` reports `PORT=80` in the working configuration (mup 1.4.6), and `PORT=3000` in the broken configuration (mup 1.5.0.beta). I imagine this change is the source of the problem, but I don't see what needs to change to fix it, as it matches what you expected.

**Comment by zodern** [comment]
There is a bug that when using the reverse proxy it exposes the wrong port. It will be fixed in the next beta.

**Comment by zodern** [comment]
This is fixed in Mup 1.5.
